src/main/java/com/google/api/codegen/config/DiscoveryField.java

```diff
@@ -29,25 +29,40 @@
 import com.google.api.codegen.util.TypeName;
 import com.google.api.codegen.util.TypeNameConverter;
 import com.google.api.tools.framework.model.Oneof;
-import com.google.api.tools.framework.model.TypeRef;
 import com.google.api.tools.framework.model.TypeRef.Cardinality;
 import com.google.common.base.Preconditions;
 import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import javax.annotation.Nullable;
 
-/** Created by andrealin on 7/31/17. */
-public class DiscoveryField implements FieldModel {
+/** A field declaration wrapper around a Discovery Schema. */
+public class DiscoveryField implements FieldModel, TypeModel {
```

Comment:
Out of curiosity why not split this into `DiscoveryField implements FieldModel` and `DiscoveryTypeRef implements TypeModel`?

Reply:
Conceptually, a FieldModel contains an instance of TypeModel. It seemed redundant to create another class that to separately implement TypeModel because the underlying Discovery Schema can take care of filling in the methods for both FieldModel and TypeModel.